src/main/java/org/vivoweb/harvester/fetch/JSONFetch.java

```diff
@@ -363,68 +382,230 @@ public void execute() throws IOException {
-       sb.append(">");
-
-       // insert field value
-       if (val instanceof  JSONArray) {
-    	   JSONArray array = (JSONArray) val;
-    	   log.debug("field is an array: "+ field);
-    	   Iterator iter = array.iterator();
-    	   while (iter.hasNext()) {
-    		   Object obj = iter.next();
-               log.debug("objtype: "+ obj.getClass().getName());
-    		   log.debug("val: "+ array.toString());
-    	   }
-       } else {
-         sb.append(SpecialEntities.xmlEncode(val.toString().trim()));
-       } 
-       // Field END
-       sb.append("</");
-       sb.append(SpecialEntities.xmlEncode(field));
-       sb.append(">\n");
-       return sb.toString();
-    }
 
+        private void arrayHandlingV2(Object val, StringBuffer sb, XMLTagIndexing xmlTagIndexing, String fixedkey) {
+            JSONArray array = (JSONArray) val;
 
+            sb.append("\n");
+            Iterator arrayIterator = array.iterator();
 
+            log.debug("val: "+ val);
 
+            while (arrayIterator.hasNext()) {
+                Object obj = arrayIterator.next();
+
+                if (!xmlTagIndexing.isArrayIndexOpen()) {
+                    xmlTagIndexing.setArrayIndexOpen();
+                    String lastChar = fixedkey.substring(fixedkey.length() - 1);
+                    if (lastChar.equals("s"))
+                        xmlTagIndexing.setXmlTagName(StringUtils.chop(fixedkey));
+                    else
+                        xmlTagIndexing.setXmlTagName(fixedkey);
+                    sb.append("    <"+xmlTagIndexing.getXmlTagName()+"_"+ xmlTagIndexing.getElementNo() +">");
+                }
+
+//            log.debug("val: "+ obj);
+
+                if (obj instanceof JSONArray) {
+                    log.debug("there is an JSON Array inside: "+ obj);
+                    XMLTagIndexing xmlArrayIndexing = new XMLTagIndexing();
+                    xmlArrayIndexing.setElementNo(0);
+
+                    arrayHandlingV2(val, sb, xmlArrayIndexing, fixedkey);
+                } else if (obj instanceof JSONObject) {
+                    log.debug("there is an JSON Object inside: "+ obj);
+                    objectHandling(obj, sb);
+                }
+                else {
+                    sb.append(obj.toString()
+                            .replaceAll("&", "&amp;")
+                            .replaceAll("</br>","")
+                            .replaceAll("<","")
+                            .replaceAll(">",""));
+                }
+                if (xmlTagIndexing.isArrayIndexOpen()) {
+                    sb.append("</"+xmlTagIndexing.getXmlTagName()+"_"+ xmlTagIndexing.getElementNo() +">\n");
+                    xmlTagIndexing.increaseElementNo();
+                    xmlTagIndexing.setArrayIndexClosed();
+                }
+            }
+            sb.append("    ");
+        }
+
+        public String getTagName(String field, Object val) {
+            StringBuffer sb = new StringBuffer();
```

Comment:
Why is `StringBuffer` used here? If this class will not be used in a multithreaded environment I think we should switch to using `StringBuilder` everywhere because it is a lot faster.